src/downloadlist.cpp

```diff
@@ -270,63 +251,194 @@ bool DownloadList::lessThanPredicate(const QModelIndex& left, const QModelIndex&
+        const MOBase::ModRepositoryFileInfo* info = m_manager.getFileInfo(lDownloadId);
         leftID                                    = info->modID;
       }
 
-      if (!m_manager.isInfoIncomplete(right.row())) {
-        const MOBase::ModRepositoryFileInfo* info = m_manager.getFileInfo(right.row());
+      if (!m_manager.isInfoIncomplete(rDownloadId)) {
+        const MOBase::ModRepositoryFileInfo* info = m_manager.getFileInfo(rDownloadId);
         rightID                                   = info->modID;
       }
 
       return leftID < rightID;
     } else if (left.column() == DownloadList::COL_STATUS) {
-      DownloadManager::DownloadState leftState  = m_manager.getState(left.row());
-      DownloadManager::DownloadState rightState = m_manager.getState(right.row());
+      DownloadManager::DownloadState leftState  = m_manager.getState(lDownloadId);
+      DownloadManager::DownloadState rightState = m_manager.getState(rDownloadId);
       if (leftState == rightState)
-        return m_manager.getFileTime(left.row()) > m_manager.getFileTime(right.row());
+        return m_manager.getFileTime(lDownloadId) >
+               m_manager.getFileTime(rDownloadId);
       else
         return leftState < rightState;
     } else if (left.column() == DownloadList::COL_SIZE) {
-      return m_manager.getFileSize(left.row()) < m_manager.getFileSize(right.row());
+      return m_manager.getFileSize(lDownloadId) < m_manager.getFileSize(rDownloadId);
     } else if (left.column() == DownloadList::COL_FILETIME) {
-      return m_manager.getFileTime(left.row()) < m_manager.getFileTime(right.row());
+      return m_manager.getFileTime(lDownloadId) <
+             m_manager.getFileTime(rDownloadId);
     } else if (left.column() == DownloadList::COL_SOURCEGAME) {
-      return m_manager.getDisplayGameName(left.row()) <
-             m_manager.getDisplayGameName(right.row());
+      return m_manager.getDisplayGameName(lDownloadId) <
+             m_manager.getDisplayGameName(rDownloadId);
     } else {
       return leftIndex < rightIndex;
     }
   } else {
     return leftIndex < rightIndex;
   }
 }
+
+Download* DownloadList::getDownload(int downloadId) {
+  auto download = std::find_if(m_downloads.begin(), m_downloads.end(),
+                      [downloadId](const Download download) {
+                        return download.downloadId == downloadId;
+                      });
+  if (download != m_downloads.end()) {
+    return download;
+  }
+
+  return nullptr;
+}
+
+int DownloadList::getDownloadIndex(int downloadId)
+{
+  auto download = std::find_if(m_downloads.begin(), m_downloads.end(),
+                               [downloadId](const Download download) {
+                                 return download.downloadId == downloadId;
+                               });
+  
+  if (download != m_downloads.end()) {
+    return download - m_downloads.begin();
+  }
+
+  return -1;
+}
+
+void DownloadList::buildDownload(Download* download) {
```

Comment:
I'll add this that since the name is very confusing and the function does different things, it's very hard to understand what's happening every time a "build download" is made. I think this function should be removed and the code move elsewhere.

Reply:
changed the naming of this function to getDownloadInfo. it's primary use is to get the latest information of a specific DownloadInfo item from DownloadManager and apply that to the DownloadListItem.